src/segment-loader.js

```diff
@@ -1328,4 +1562,144 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     this.segmentMetadataTrack_.addCue(cue);
   }
+
+  // TODO: documentation
+  maxBufferedBytes(start, end) {
+    if (!this.segmentMetadataTrack_) {
+      return 0;
+    }
+
+    let i;
+    let j;
+    let cue;
+    let cmpStart;
+    let cmpEnd;
+    let bytes = 0;
+    let buffered = this.buffered_();
+
+    start = start === undefined ? 0 : start;
+    end = end === undefined ? Infinity : end;
+
+    for (i = 0; i < this.segmentMetadataTrack_.cues.length; i++) {
+      cue = this.segmentMetadataTrack_.cues[i];
+      for (j = 0; j < buffered.length; j++) {
+        cmpStart = (start > buffered.start(j)) ? start : buffered.start(j);
```

Comment:
I find this sort of confusing. How/why are we adjusting the start cmpStart and cmpEnd variables here?